### PR TITLE
[stable/filebeat] Allow for additional operator overrides

### PR DESCRIPTION
**Pull request by marc-sensenich**
**What this PR does / why we need it**:
Allows for users to add `command`, `args`, `imagePullSecrets`, and `envVars` from secrets and configMaps

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/filebeat/templates/daemonset.yaml [hunk 1]
```diff
@@ -28,6 +28,12 @@ spec:
 {{ toYaml .Values.annotations | indent 8 }}
 {{- end }}
     spec:
+    {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range $sec := .Values.image.pullSecrets }}
+        - name: {{ $sec | quote }}
+      {{- end }}
+      {{- end }}
```
Comment (davidkarlsen) [inline stable/filebeat/templates/daemonset.yaml]: align end with if

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: stable/filebeat/README.md [hunk 2]
```diff
@@ -40,8 +41,10 @@ The following table lists the configurable parameters of the filebeat chart and
...
+| `command`                                                | Custom command (Docker Entrypoint)                                                                       | `[]`                                               |
+| `args`                                                   | Custom args (Docker Cmd)                                                                                 | `[]`                                               |
 | `plugins`                                                | List of beat plugins                                                                                     | `[]`                                               |
-| `extraVars`                                              | A map of additional environment variables                                                                | `{}`                                               |
+| `extraVars`                                              | A list of additional environment variables                                                                | `[]`                                              |
```
Comment (davidkarlsen) [inline stable/filebeat/README.md:47]: this is a breaking change as the format changes - is it worth it?
if it is - it means a new major version of the chart

File: stable/filebeat/values.yaml [hunk 3]
```diff
@@ -57,9 +57,21 @@ indexTemplateLoad: []
...
+# pass custom args. This is equivalent of Cmd in docker
+args: []
+
+# A list of additional environment variables
+extraVars: []
+  # - name: TEST1
+  #   value: TEST2
+  # - name: TEST3
+  #   valueFrom:
+  #     configMapKeyRef:
```
Comment (davidkarlsen) [inline stable/filebeat/values.yaml:72]: ah - I see now why you want the list format

File: stable/filebeat/templates/daemonset.yaml (hunk 1 above)
Comment (marc-sensenich) [reply stable/filebeat/templates/daemonset.yaml]: Addressed

File: stable/filebeat/values.yaml (hunk 3 above)
Comment (marc-sensenich) [reply stable/filebeat/values.yaml:72]: Yeah, there are values for our deployment that will be loaded from secrets that would be accessed via this manner.

File: stable/filebeat/Chart.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 description: A Helm chart to collect Kubernetes logs with filebeat
 icon: https://www.elastic.co/assets/blt47799dcdcf08438d/logo-elastic-beats-lt.svg
 name: filebeat
-version: 0.8.0
```
Comment (davidkarlsen) [inline stable/filebeat/Chart.yaml:5]: Since parameter format changes I think the major version needs to change. 1.0.0

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: stable/filebeat/Chart.yaml (hunk 4 above)
Comment (cpanato) [reply stable/filebeat/Chart.yaml:5]: Also in the charts meeting we discussed about bump the chart version for stable charts for at least 1.0.0 of those are in 0.x.x

Comment (marc-sensenich) [reply stable/filebeat/Chart.yaml:5]: Bumped the version to 1.0.0

**Review by davidkarlsen — APPROVED** [review]

**Pull request merged**